docs/generate_cmd_doc.py

```diff
@@ -0,0 +1,182 @@
+import logging
+import os
+import re
+import subprocess
+import typing
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+AUTOMATED_START_MARKER = "<!-- START AUTOMATED OUTPUT -->"
+AUTOMATED_STOP_MARKER = "<!-- END AUTOMATED OUTPUT -->"
+
+TEMPLATE = """\
+(reference_pebble_cmd_command)=
+# cmd command
+
+Description
+
+## Usage
+
+<!-- START AUTOMATED OUTPUT -->
+```{terminal}
+   :input: command
+```
+<!-- END AUTOMATED OUTPUT -->
+"""
+
+
+def get_all_pebble_commands() -> typing.List[str]:
+    cmd = "pebble help --all"
+    p = subprocess.run(cmd.split(), text=True, capture_output=True)
+    if p.returncode != 0:
+        logger.error("Error running {}.".format(cmd))
+        exit(1)
+    return re.findall(r"\n\s{4}(\w+)", p.stdout)
+
+
+def get_command_help_output(cmd: str) -> str:
+    p = subprocess.run(cmd.split(), text=True, capture_output=True)
+    if p.returncode != 0:
+        logger.error("Error running {}.".format(cmd))
+        exit(1)
+    return p.stdout
+
+
+def get_description_from_output(text: str) -> str:
+    pattern = r"Usage:\n.*?\n\n(.*?)\n.*"
+    match = re.search(pattern, text, re.DOTALL)
+    if match:
+        desired_block = match.group(1).strip()
+        return desired_block
+    return ""
+
+
+def render_title(text: str, cmd: str) -> str:
+    text = re.sub(
+        r"\(reference_pebble_cmd_command\)=",
+        f"(reference_pebble_{cmd}_command)=",
+        text,
+        flags=re.MULTILINE,
+    )
+    return re.sub(
+        r"^# cmd command$",
+        f"# {cmd} command",
+        text,
+        flags=re.MULTILINE,
+    )
+
+
+def render_description(text: str, description: str) -> str:
+    return re.sub(r"^Description$", f"{description}", text, flags=re.MULTILINE)
+
+
+def render_code_block_cmd(text: str, cmd: str) -> str:
+    return re.sub(r"(:input: ).*$", rf"\1{cmd}", text, count=1, flags=re.MULTILINE)
+
+
+def render_code_block_output(text: str, output: str) -> str:
+    start_pos = text.find(AUTOMATED_START_MARKER)
+    end_pos = text.find(AUTOMATED_STOP_MARKER) + len(AUTOMATED_STOP_MARKER)
+    return text[:start_pos] + output + text[end_pos:]
+
+
+def update_toc(all_cmds: typing.List[str]):
+    index_page = "reference/cli-commands/cli-commands.md"
+    with open(index_page, "r") as file:
+        text = file.read()
+
+    start_index = text.find("```{toctree}")
+    end_index = text.find("```", start_index + 1) + 3
+    cmd_list = "\n".join(["{cmd} <{cmd}>".format(cmd=cmd) for cmd in all_cmds])
```

Comment:
Can we use f-strings? You've sometimes used f-strings in this PR, sometimes `str.format`. Let's be consistent. You can also drop the extra `[]` brackets and use a generator expression directly in the `join`:

```python
cmd_list = '\n'.join(f'{cmd} <{cmd}>' for cmd in all_cmds)
```